internal/cli/cmd/auth/login.go

```diff
@@ -91,3 +82,72 @@ func openURL(url string) bool {
 	err := browser.OpenURL(url)
 	return err == nil
 }
+
+type tenant struct {
+	token string
+	name  string
+}
+
+func completeLogin(ctx context.Context, id, kind string) (tenant, error) {
+	ephemeralCliId := fnapi.TelemetryOn(ctx).GetClientID()
+
+	if kind == "tenant" {
+		res, err := fnapi.CompleteTenantLogin(ctx, id, kind)
+		if err != nil {
+			return tenant{}, err
+		}
+
+		return tenant{
+			token: res.TenantToken,
+			name:  res.TenantName,
+		}, nil
+	}
+
+	// TODO remove old login path
+	userAuth, err := getUserAuth(ctx, id, kind, ephemeralCliId)
+	if err != nil {
+		return tenant{}, err
+	}
+
+	if _, err := auth.StoreUser(ctx, userAuth); err != nil {
+		return tenant{}, err
+	}
+
+	userToken, err := auth.GenerateTokenFromUserAuth(ctx, userAuth)
+	if err != nil {
+		return tenant{}, err
+	}
+
+	tt, err := fnapi.ExchangeUserToken(ctx, userToken)
+	if err != nil {
+		return tenant{}, err
+	}
+
+	return tenant{token: tt.TenantToken}, nil
+}
+
+func getUserAuth(ctx context.Context, id, kind, ephemeralCliId string) (*auth.UserAuth, error) {
+	switch kind {
+	case "tenant":
```

Comment:
This branch is not exercised? Remove?

Reply:
Well, it is defensive coding. If you don't mind, I'd keep it :)